src/Coupon/WCCouponAdapter.php

```diff
@@ -267,6 +267,20 @@ protected function map_wc_usage_restriction( WC_Coupon $wc_coupon ): WCCouponAda
 			$this->setProductTypeExclusion( $str_product_categories );
 		}
 
+		$wc_product_ids_in_brand = $this->get_product_ids_in_brand( $wc_coupon );
+		if ( ! empty( $wc_product_ids_in_brand ) ) {
+			$google_product_ids      = array_map( $get_offer_id, $wc_product_ids_in_brand );
+			$has_product_restriction = true;
+			$this->setItemId( $google_product_ids );
+		}
+
+		$wc_excluded_product_ids_in_brand = $this->get_product_ids_in_brand( $wc_coupon, true );
+		if ( ! empty( $wc_excluded_product_ids_in_brand ) ) {
+			$google_product_ids      = array_map( $get_offer_id, $wc_excluded_product_ids_in_brand );
+			$has_product_restriction = true;
+			$this->setItemIdExclusion( $google_product_ids );
```

Comment:
When a coupon also has other excluded products, those product IDs will be overridden by this restriction.

For example, the brand in the following setting is only added to a t-shirt, the restriction synced to GMC is only the product ID of that t-shirt, while the product ID of the belt does not appear.

![image](https://github.com/user-attachments/assets/6f660ffd-5c1e-4a90-890a-6e1d29a855ab)

However, when tested in WC store, both restrictions are applied.

---

I also tested the opposite restrictions and looks like the brand inclusion restriction has a higher priority than the product inclusion restriction, so the overriding for inclusion restriction should be fine.

Reply:
Nice catch!

I believe it could potentially be a bug in Brands that product inclusion being overridden by the product brands inclusion:

Brands [uses a filter for checking product brands inclusion](https://github.com/woocommerce/woocommerce/blob/5a182ac082b83ab0642199f04b030b04c3f1ade4/plugins/woocommerce/includes/class-wc-brands-coupons.php#L42-L63) after [the product inclusion validation](https://github.com/woocommerce/woocommerce/blob/5a182ac082b83ab0642199f04b030b04c3f1ade4/plugins/woocommerce/includes/class-wc-discounts.php#L1056), and it did not take it into account.

The filter has a third parameter which is a discount object, I think it might be ok to recheck the product inclusion in Brands.

However, to align with the current WC core behaviour, I will update this PR to respect only the product exclusion, and will also create an issue in WC core for Brands about the inclusion bug.

Reply:
Issue created in woocommerce/woocommerce/issues/53299.